Framework/PythonInterface/plugins/algorithms/MaskAngle.py

```diff
@@ -48,61 +50,58 @@ def PyInit(self):
                              mantid.kernel.StringListValidator(['TwoTheta', 'Phi']),
                              'Which angle to use')
         self.declareProperty(mantid.kernel.IntArrayProperty(name="MaskedDetectors", direction=mantid.kernel.Direction.Output),
-                             doc="List of detector masked, with scatterin angles between MinAngle and MaxAngle")
+                             doc="List of detector masked, with scattering angles between MinAngle and MaxAngle")
 
     def validateInputs(self):
         issues = dict()
-
         ws = self.getProperty("Workspace").value
-
-        try:
-            if type(ws).__name__ == "WorkspaceGroup":
-                for w in ws:
-                    w.getInstrument().getSource().getPos()
-            else:
-                ws.getInstrument().getSource().getPos()
-        except (RuntimeError, ValueError, AttributeError, TypeError):
+        hasInstrument = True
+        if type(ws).__name__ == "WorkspaceGroup" and len(ws) > 0:
+            for item in ws:
+                hasInstrument = hasInstrument and len(item.componentInfo()) > 0
+        else:
+            hasInstrument = len(ws.componentInfo()) > 0
+        if not hasInstrument:
             issues["Workspace"] = "Workspace must have an associated instrument."
-
         return issues
 
+    def _get_phi(self, spectra_pos):
+        '''
+        The implementation here assumes that z is the beam direction.
+        That assumption is not universally true, it depends on the geometry configuration.
+        This returns the phi spherical coordinate value
+        '''
+        return math.fabs(math.atan2(spectra_pos.Y(), spectra_pos.X()))
```

Comment:
Did we finally came up with a solution to get the phi angle in a "universal" way? I do not want to implement the same function everywhere where I need phi in Instrument 2.0

Reply:
Also, no need to import math. You already imported numpy

Reply:
We can probably come up with some reasonable default parameter names and values, so when calling `getPhi` with no parameters you get the current behavior.